File: Kaskus/GridViewPaging/GridViewPagingVb/Datas/RepositoryItemGrid.vb
Imports System.Globalization
Imports Dede.App.GridViewPaging.Domains
Imports Dede.App.GridViewPaging.Pages

Namespace Datas

    Public Class RepositoryItemGrid
        Implements IDataSourceRepository

        Private Shared ReadOnly ItemsGrids As New List(Of ItemGrid)()

        Public Function GetData(pageSize As Int32, currentPage As Int32) _
            As DataSourceResult _
            Implements IDataSourceRepository.GetData

            InitItemGrid()

            Return New DataGenericResult(Of ItemGrid)() With { _
                .Records = ItemsGrids.Skip(currentPage).Take(pageSize).ToList(), _
                .TotalRecord = ItemsGrids.Count }
        End Function

        Private Shared Sub InitItemGrid()
            If Not ItemsGrids.Count.Equals(0) Then
                Return
            End If

            ItemsGrids.Clear()
            ItemsGrids.AddRange(GenerateItemGrid())
        End Sub

        Private Shared Function GenerateItemGrid() As IEnumerable(Of ItemGrid)
            Dim rnd = New Random(100000)
            Dim dat = DateTime.Now

            Dim list = New List(Of ItemGrid)

            For i As Int32 = 0 To 1000 - 1
                Dim itemRandom = rnd.Next()
                Dim modular = itemRandom Mod 2
                Dim dec = itemRandom * 0.2D
                dat = dat.AddDays(0 - (itemRandom / CType(dec, Int32)))

                list.Add(New ItemGrid() With { _
                    .Id = Guid.NewGuid(), _
                    .ItemBoolean = modular = 0, _
                    .ItemDateTime = dat, _
                    .ItemDecimal = dec, _
                    .ItemInt32 = i, _
                    .ItemString = itemRandom.ToString(CultureInfo.InvariantCulture)})
            Next

            Return list
        End Function
    End Class
End Namespace

File: Kaskus/GridViewPaging/GridViewPagingCs/Datas/RepositoryItemGrid.cs
namespace Dede.App.GridViewPaging.Datas
{
    using System;
    using System.Collections.Generic;
    using System.Globalization;
    using System.Linq;

    using Dede.App.GridViewPaging.Domains;
    using Dede.App.GridViewPaging.Pages;

    public sealed class RepositoryItemGrid
        : IDataSourceRepository
    {
        private static readonly List<ItemGrid> ItemsGrids = new List<ItemGrid>();

        public DataSourceResult GetData(
            Int32 pageSize, 
            Int32 currentPage)
        {
            InitItemGrid();

            return new DataGenericResult<ItemGrid>
            {
                Records = ItemsGrids
                    .Skip(currentPage)
                    .Take(pageSize)
                    .ToList(),
                TotalRecord = ItemsGrids.Count
            };
        }

        private static void InitItemGrid()
        {
            if (!ItemsGrids.Count.Equals(0))
            {
                return;
            }

            ItemsGrids.Clear();
            ItemsGrids.AddRange(GenerateItemGrid().ToList());
        }

        private static IEnumerable<ItemGrid> GenerateItemGrid()
        {
            var rnd = new Random(100000);
            var dat = DateTime.Now;

            for (var i = 0; i < 1200; i++)
            {
                var itemRandom = rnd.Next();

                var mod = itemRandom % 2;
                var dec = itemRandom * 0.2M;
                dat = dat.AddDays(0 - (itemRandom / (Int32)dec));

                yield return
                    new ItemGrid
                    {
                        Id           = Guid.NewGuid(),
                        ItemBoolean  = mod == 0,
                        ItemDateTime = dat,
                        ItemDecimal  = dec,
                        ItemInt32    = i,
                        ItemString   = itemRandom.ToString(CultureInfo.InvariantCulture)
                    };
            }
        }
    }
}